content/blog/2020-12-19-LLVM-IR-to-assembly.md

```diff
@@ -0,0 +1,76 @@
+
+
+## Microbenchmark w
+
+### Process and Methology
+The microbenchmark preforms a series of 4906 4x4 matrix multiplications.  A total of 4096 iterations have been preformed, and the results are aggregated by the minimum function.  Due to the small amount of cycle it takes to complete the operations, using wall-clock time may introduce another source of error, I have elected to report the results in CPU cycles.
+
+### Results
+Table 1, Microbenchmark of a series 4096 4x4 matrix in CPU Cycles, 4096 trials, reporting average minimum number cycles required to complete the a single 4x4 matrix multiplication. 
+
+| Microbenchmark 4x4   matrix             | GCC                   | Clang  | Clang (0.75) |
+|-----------------------------------------|-----------------------|--------|--------------|
+| REF                                     | 227.76                | 214.98 | 214.8        |
+| SSE                                     | 144.06                | 124.48 | 124.33       |
+| AV_4mem                                 | 140.11                | 170.03 | 170.14       |
+| AVX_8                                   | 93.48                 | 98.99  | 95.75        |
+
+### Results and Analysis
+As this microbenchmark is designed to measure the maximum performance of the complied code, I have chosen to aggregate the result with a min function. Preliminary analysis has shown that reducing the number of architectural registers has little or no effect on the performance on microbenchmarks.  This is as expected as microbenchmarks do not have high amounts of register pressure as they often contain couple of register operands.  In this case, the compiler backend implementation has more of an impact on the final performance than the modification of the number of x86-64 architectural registers.  
+
+
+## FFT benchmark
+
+### Process and Methology
+The benchmark preforms a set of FFT calculations.  A total of 6 iterations have been performed and the data was preformed statistical analysis (Confidenace interval).  Execution time is measured in wall-clock time.
+
+### Results
+Table 2, FFT benchmark in wall-clock time, average number secounds required to complete the operation.  
+| Exec time (s) | GCC         | Clang       | Clang (0.75) |
+|---------------|-------------|-------------|--------------|
+| Average       | 2833314     | 2548542     | 3081897      |
+| Min           | 2522625     | 2456454     | 2643296      |
+| Std. Dev      | 381923      | 66116       | 497256       |
+| CI-Lower(0.01)| 2431691     | 2479015     | 2558993      |
+| CI-Upper(0.01)| 3234936     | 2618069     | 3604801      |
+| CI-Lower(0.05)| 2527717     | 2495639     | 2684016      |
+| CI-Upper(0.05)| 3138911     | 2601446     | 3479778      |
```

Comment:
I'm still confused about the units here. You say "average number secounds," but that's a *lot* of seconds! 2833314 seconds is 33 days, for example.